### DISCUSSION TITLE
‘Parameterization + texturing from registered rasters’ filter without using set_raster_camera

### DISCUSSION THREAD

**Discussion by TaraRF** [cnr-isti-vclab/PyMeshLab#123] [Q&A]
Hi, 
I've been using MeshLab to texture meshes by using the 'texture>Parameterization + texturing from registered rasters' method. I want to automate this process using PyMeshLab.

It seems that I should use 'set_raster_camera' filter first before using 'Parameterization + texturing from registered rasters' filter. But I see in the PyMeshLab documentation that 'shotf', parameter used in this filter, is still unsupported. We solved this problem by tweaking the project’s .mlp file, making sure that mesh and raster are fully aligned. When I load this project in MeshLab, and directly apply 'Parameterization + texturing from registered rasters’ filter, it creates a decent texture. When I load this project in PyMeshLab, and then apply the Parameterization filter, I get completely different texture. I wonder what is causing this difference.

Thanks!

**Comment by alemuntoni** [comment]
Hi,
could be some bug in the code... Could you please share a mlp, with mesh and rasters that cause the problem?

**Comment by TaraRF** [comment]
I am afraid I cannot share the data due to its sensitivity. I could perhaps create a dummy data if it helps. The mlp file looks like this:
```
<!DOCTYPE MeshLabDocument>
	<MeshLabProject>
		<MeshGroup>
			<MLMesh visible="1" idInFile="-1" label="mesh.ply" filename="mesh.ply">
				<MLMatrix44>        1 0 0 0         0 1 0 0         0 0 1 0         0 0 0 1         </MLMatrix44>
				<RenderingOption pointColor="131 149 69 255" pointSize="3" wireWidth="1" wireColor="64 64 64 255" solidColor="192 192 192 255" boxColor="234 234 234 255">100001010000000000000100000001011110000010100000000100111010100000001001</RenderingOption>
			</MLMesh>
		</MeshGroup>
		<RasterGroup>
			<MLRaster label="../data/0-color.png">
				<VCGCamera FocalMm="90" ViewportPx="640 480" TranslationVector="0 0 0.5258900380392899 1" BinaryData="0" LensDistortion="0 0" CameraType="0" RotationMatrix="1 0 0 0 0 -1 0 0 0 0 -1 0 0 0 0 1" PixelSizeMm="0.0967509 0.0967509" CenterPx="320 240"/>
				<Plane fileName="../data/0-color.png" semantic="1"/>
			</MLRaster>
		</RasterGroup>
	</MeshLabProject>
```
The code in PyMeshLab: 
```
ms = ml.MeshSet()
ms.load_project(project_file_path)
ms.parameterization__texturing_from_registered_rasters(texturename=output_texture_path)
ms.save_current_mesh(output_mesh_path)
```
Meanwhile, we figured out that by upgrading PyMeshLab from 0.2 to 0.2.1, we could solve the problem in Windows.  Meshlab version is 2021.05. Now both textures (coming from PyMeshLab as well as MeshLab) are identical. 
The code still gives an error in the Parameterization line when we run it in WSL in Windows as well as a Linux machine (Ubuntu 18).
It is worth mentioning that PyMeshLab in Ubuntu performs well when I apply other filters such as surface_reconstruction_screened_poisson. Only Parameterization filter has had this issue so far.

**Reply by alemuntoni** [reply]
Ok, maybe it's worth to first update to the last version of pymeshlab, which is 2021.07.

Then, please share the output error that you get when you apply the filters and also some dummy data that helps me on replicating the issue. I need as much info as possible to investigate and then solve issues...

I'll try to make available the shotf parameter in the next version of pymeshlab.
